src/components/modules/blockManager.ts

```diff
@@ -471,7 +471,12 @@ export default class BlockManager extends Module {
    * @returns {Promise} - the sequence that can be continued
    */
   public async mergeBlocks(targetBlock: Block, blockToMerge: Block): Promise<void> {
-    const blockToMergeData = await blockToMerge.data;
+    const blockToMergeData = targetBlock.name !== blockToMerge.name
+      ? convertStringToBlockData(
+        await blockToMerge.exportDataAsString(),
+        targetBlock.tool.conversionConfig
+      )
+      : await blockToMerge.data;
 
     if (!_.isEmpty(blockToMergeData)) {
       await targetBlock.mergeWith(blockToMergeData);
```

Comment:
I see an error trying to merge Paragraph and Nested List:

![image](https://github.com/codex-team/editor.js/assets/3684889/85f06b5e-83eb-4ca9-8bdf-ccfeefcc998a)

Probably, in case of conversion, we should check if target block has the `merge` method (`.mergeable`):
- if yes, call `mergeWith`
- if not, use `convert()` method:

```ts
this.convert(targetBlock, targetBlock.name, blockToMergeData)
```

Reply:
That might be the reason I was checking with `blockToMerge.mergeable` in the `areBlocksMergeable` function

Reply:
But from what I understand with the `this.convert` it's going to override the previous block value 🤔 

This is not what we want, no?

Reply:
I meant technically we can merge two blocks if they have `conversion` config without calling a tool's `merge()` method. That's why `mergeable` was not required for this case.

But you're right. In that case, there will be a block replacing. Internally the `convert()` removes the previous block and adds the new one. So the `onChange` callback will get two events: `block-removed` and `block-added`. It will be invisible for end-user, but it could be a problem for some developers relying on that events since expected ones are `block-removed` and `block-changed`.

Thus, I'd suggest to require `merge` for merging, like you have previously.